python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -133,19 +109,50 @@ cdef class GroupBy:
         from cudf.core.column_accessor import ColumnAccessor
         cdef vector[libcudf_groupby.aggregation_request] c_agg_requests
         cdef Column col
+        cdef Aggregation agg_obj
 
-        aggregations = _drop_unsupported_aggs(values, aggregations)
+        allow_empty = all(len(v) == 0 for v in aggregations.values())
 
+        included_aggregations = defaultdict(list)
+        idx = 0
         for i, (col_name, aggs) in enumerate(aggregations.items()):
             col = values._data[col_name]
+            dtype = col.dtype
+
+            valid_aggregations = (
+                _LIST_AGGS if is_list_dtype(dtype)
+                else _STRING_AGGS if is_string_dtype(dtype)
+                else _CATEGORICAL_AGGS if is_categorical_dtype(dtype)
+                else _STRING_AGGS if is_struct_dtype(dtype)
+                else _INTERVAL_AGGS if is_interval_dtype(dtype)
+                else _DECIMAL_AGGS if is_decimal_dtype(dtype)
+                else "ALL"
+            )
```

Comment:
Given that `dtype` is guaranteed to be an instance of a known type, do you think we can replace this with a lookup from a dict with types (such as `ListDtype`) as keys?

Reply:
How would you map the boolean outputs of the various `is_*` functions to the keys?

Reply:
I guess I was picturing something like:

```python
dtype_to_aggs = {ListDtype: _LIST_AGGS, np.dtype("object"): _STRING_AGGS, ...}
valid_aggregations = _dtype_to_aggs[dtype]
```

Reply:
Ah I see what you mean by a known type, you can guarantee the input will be such that you won't need to worry about some of the more esoteric cases handled by these different functions. In any case, let's hold off on this and see if it can be subsumed by some of the broader dtype changes I've proposed.